panic: runtime error: index out of range

panic: runtime error: index out of range

goroutine 1 [running]:
gadb/src/github.com/lsl/gadb.readDevices(0x11c66f0, 0x11c66e0, 0x11c66e0)
	/gadb/src/github.com/lsl/gadb/gadb.go:87 +0x73a
gadb/src/github.com/lsl/gadb.Gadb()
	/gadb/src/github.com/lsl/gadb/gadb.go:112 +0x34
main.main()
	gadb/main.go:6 +0x20

--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 BINARY_NAME=gadb
-VERSION=1.0.0
+VERSION=1.0.3
 build:
 
 	rm -rf *.gz

--- a/src/github.com/lsl/gadb/gadb.go
+++ b/src/github.com/lsl/gadb/gadb.go
@@ -84,6 +84,9 @@ func readDevices() []devices {
 		s := strings.Trim(line, "\n")
 		if !strings.HasPrefix(s, "List of devices") && s != "" {
 			var ss = re.Split(s, -1)
+			if ss[1] == "offline" {
+				continue
+			}
 			var dv = devices{ss[0], ss[2], ss[3], ss[4], ss[5]}
 			arrays = append(arrays, dv)
 		}

